nbconvert/exporters/templateexporter.py

```diff
@@ -217,9 +223,26 @@ def _load_template(self):
         # template by name with extension added, then try loading the template
         # as if the name is explicitly specified.
         template_file = self.template_file
-        self.log.debug("Attempting to load template %s", template_file)
-        self.log.debug("    template_path: %s", os.pathsep.join(self.template_path))
-        return self.environment.get_template(template_file)
+        
+        try:
+            self._log_template_loading(template_file)
+            return self.environment.get_template(template_file)
+        except TemplateNotFound:
+            if self.template_file.endswith(self.template_extension):
+                try: 
+                    template_file = self.template_file[:-len(self.template_extension)]
+                    self._log_template_loading(template_file)
+                    return self.environment.get_template(template_file)
+                except TemplateNotFound:
+                    raise TemplateNotFound(template_file)
+            else:
+                try:
+                    template_file = self.template_file + self.template_extension
+                    self._log_template_loading(template_file)
+                    return self.environment.get_template(template_file)
+                except TemplateNotFound:
+                    raise TemplateNotFound(template_file)
```

Comment:
Having a bunch of checks and fallbacks here is precisely what I was trying to avoid by making the ExtensionTolerantLoader, so I'd rather not put them all back in here.